hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+                if c.group_name != "CZO National":  # The National Group is used to establish the entire Community
+                    res_count = len([r for r in community_resources if r.group_name == c.group_name])
+                    groups.append({'id': str(c.group_id), 'name': str(c.group_name), 'res_count': str(res_count)})
+
+        groups = sorted(groups, key=lambda key: key['name'])
+        return {
+            'community_resources': community_resources,
+            'groups': groups,
+            'grpfilter': grpfilter,
+        }
+
+
+class FindCommunitiesView(TemplateView):
+    template_name = 'hs_communities/find-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(FindCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+
+        return {
+            'communities_list': Community.objects.all()
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class MyCommunitiesView(TemplateView):
+    template_name = 'hs_communities/my-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(MyCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def group_to_community(self, grp, communities):
+        """
+        return the community membership information of a group; group can belong to only one community
+        :param grp: group object
+        :param communities: communities
+        :return: tuple id, name of community
+        """
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    def get_context_data(self, **kwargs):
+        grps_member_of = []
+        u = User.objects.get(pk=self.request.user.id)
+        groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
+        # for each group set group dynamic attributes
+        for g in groups:
+            g.is_user_member = u in g.gaccess.members
+            if g.is_user_member:
+                grps_member_of.append(g)
+            g.join_request_waiting_owner_action = g.gaccess.group_membership_requests.filter(request_from=u).exists()
+            g.join_request_waiting_user_action = g.gaccess.group_membership_requests.filter(invitation_to=u).exists()
+            g.join_request = None
+            if g.join_request_waiting_owner_action or g.join_request_waiting_user_action:
+                g.join_request = g.gaccess.group_membership_requests.filter(request_from=u).first() or \
+                                 g.gaccess.group_membership_requests.filter(invitation_to=u).first()
+
+        comms_member_of = [self.group_to_community(g, Community.objects.all()) for g in grps_member_of]
+        return {
+            'communities_list': [c for c in comms_member_of if c is not None]
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class TopicsView(TemplateView):
+    """
```

Comment:
There is a requirement to limit this section to a list of `Users` by `Name` I need to discuss this with Alva/Scott for everyone's preferences how to accomplish this

Reply:
I'm still working through things but I'm starting to get an idea of what's going on.  Topic is tied to the title assistant, right?  If so, why not allow a community to each have their own set of topics as well as give the community the ability to enable/disable the title assistant.  Then the topics view could be available for users in a community with the title assistant enabled.  

I guess that raises an issue when a user belongs to multiple communities though.

Reply:
The scope of this project is not to generalize for all Communities, as the requirements for such Communities haven't been gathered. It was decided this would be scope creep.

Reply:
Not sure we're talking about the same things here... I haven't written code to administer communities or assign specific attributes to communities, based on their ID. I think the only concern here is to ensure that Topics is listed for the CZO Community only and not for all HydroShare users. Allowing another Community to have topics is probably not a good use of engineering time, as it's unknown how that Community would use Topics or if their preference is to share the edits/updates with CZO or not.

Reply:
The only Topics editor is tied to the only Title Assistant. No plans to generalize as a part of this PR.

Reply:
In your assessment, how much work would it be to split the title assistant into it's own web app?

Reply:
Far more than is budgeted for the project

Reply:
That's not an answer to my question.

Reply:
Resolving for now as it has been decided Topics will exist as global for now.